cmd/kubeapps-apis/server/packages.go

```diff
@@ -118,6 +144,119 @@ func (s packagesServer) GetAvailablePackageDetail(ctx context.Context, request *
 	}, nil
 }
 
+// GetInstalledPackageSummaries returns the installed package summaries based on the request.
+func (s packagesServer) GetInstalledPackageSummaries(ctx context.Context, request *packages.GetInstalledPackageSummariesRequest) (*packages.GetInstalledPackageSummariesResponse, error) {
+	contextMsg := ""
+	if request.Context != nil {
+		contextMsg = fmt.Sprintf("(cluster=[%s], namespace=[%s])", request.Context.Cluster, request.Context.Namespace)
+	}
+
+	log.Infof("+core GetInstalledPackageSummaries %s", contextMsg)
+
+	// Aggregate the response for each plugin
+	pkgs := []*packages.InstalledPackageSummary{}
+	// TODO: We can do these in parallel in separate go routines.
+	for _, p := range s.plugins {
+		response, err := p.server.GetInstalledPackageSummaries(ctx, request)
+		if err != nil {
+			return nil, status.Errorf(codes.Internal, "Invalid GetInstalledPackageSummaries response from the plugin %v: %v", p.plugin.Name, err)
+		}
+
+		// Add the plugin for the pkgs
+		pluginPkgs := response.InstalledPackageSummaries
+		for _, r := range pluginPkgs {
+			if r.InstalledPackageRef == nil {
+				r.InstalledPackageRef = &packages.InstalledPackageReference{}
+			}
+			r.InstalledPackageRef.Plugin = p.plugin
+		}
+		pkgs = append(pkgs, pluginPkgs...)
+	}
+
+	// Build the response
+	// TODO: Sort via default sort order or that specified in request.
+	return &packages.GetInstalledPackageSummariesResponse{
+		InstalledPackageSummaries: pkgs,
+	}, nil
+}
+
+// GetInstalledPackageDetail returns the package versions based on the request.
+func (s packagesServer) GetInstalledPackageDetail(ctx context.Context, request *packages.GetInstalledPackageDetailRequest) (*packages.GetInstalledPackageDetailResponse, error) {
+	contextMsg := ""
+	if request.InstalledPackageRef != nil && request.InstalledPackageRef.Context != nil {
+		contextMsg = fmt.Sprintf("(cluster=[%s], namespace=[%s])", request.InstalledPackageRef.Context.Cluster, request.InstalledPackageRef.Context.Namespace)
+	}
+
+	log.Infof("+core GetInstalledPackageDetail %s", contextMsg)
+
+	if request.InstalledPackageRef == nil || request.InstalledPackageRef.Plugin == nil {
+		return nil, status.Errorf(codes.InvalidArgument, "Unable to retrieve the plugin (missing InstalledPackageRef.Plugin)")
+	}
+
+	// Retrieve the plugin with server matching the requested plugin name
+	pluginWithServer := s.getPluginWithServer(request.InstalledPackageRef.Plugin)
+	if pluginWithServer == nil {
+		return nil, status.Errorf(codes.Internal, "Unable get the plugin %v", request.InstalledPackageRef.Plugin)
+	}
+
+	// Get the response from the requested plugin
+	response, err := pluginWithServer.server.GetInstalledPackageDetail(ctx, request)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Unable get the GetInstalledPackageDetail from the plugin %v: %v", pluginWithServer.plugin.Name, err)
```

Comment:
Why not just return the error, rather than creating a new one? I mean, if the plugin responded with a `NotFound` because the request was for something that had been deleted, we'd want to respond with that no? If you really see a need to wrap the text of the error, you could still extract the status code from the error returned by the plugin and use that?